lib/client.js

```diff
@@ -0,0 +1,156 @@
+var WS = require('ws');
+var debug = require('debug')('docker-exec-websocket-server:lib:client');
+var url = require('url');
+var events = require('events');
+var _ = require('lodash');
+var assert = require('assert');
+var msgcode = require('../lib/messagecodes.js');
+var through = require('through');
+
+// Makes a new buffer with code number (ideally 0x00-0xFF)
+// appended to the start of the given buffer
+function muxBuffer (code, buffer) {
+  return Buffer.concat([new Buffer([code]), buffer]);
+}
+
+class DockerExecWebsocketClient extends events.EventEmitter {
+  constructor () {
+    super();
+  }
+
+  /* Makes a client program with unbroken stdin, stdout, stderr streams
+   * is also an EventEmitter with 'exit' event
+   *
+   * Required options:
+   * url parts (hostname port pathname)
+   * or url
+   * tty: whether or not we expect VT100 style output
+   * command: array or string of command to be run in exec
+   * not supported yet: ssl cert
+   */
+  static async createClient (options) {
```

Comment:
Can't we give the options in the constructor... and then have one async method called `execute` that we call when we want to start things...

This makes it hard to see what is public and what is private properties. I know I recommended this pattern before, but in hindsight I'm not sure it's the best...